tracetool/README.md

```diff
@@ -0,0 +1,87 @@
+# Linux trace tool
+
+
+## Introduction
+
+This is python tool helps to analysis media stack trace logs combining ftrace events from libva, media driver and Linux kernel mode driver (e.g. i915).
+
+
+## Linux trace capture
+
+1. Install trace-cmd:
+
+        sudo apt-get install trace-cmd
+
+2. Grant write access to trace node for application:
+
+        sudo chmod 777 /sys/kernel/debug/
+        sudo chmod 777 /sys/kernel/debug/tracing/
+        sudo chmod 777 /sys/kernel/debug/tracing/trace_marker_raw
+
+3. Enable libva trace:
+
+        export LIBVA_TRACE = FTRACE
+
+    to enable libva buffer data capture
+
+        export LIBVA_TRACE_BUFDATA = 1
+
+4. Run application under trace-cmd in a proxy mode:
+
+        trace-cmd record -e i915 <workflow-cmd-line>
+
+5. Output is "trace.dat"
+
+Alternatively you can collect trace data in separate terminal.
+It is useful if you want to profile daemon or a service:
+
+1. Start trace capture:
+
+        sudo trace-cmd record -e i915
+
+2. Run test app in another terminal
+3. Stop capturing in the first terminal
+4. Output is "trace.dat"
+
+
+## Trace post-processing and analysis
+
+    python3 main.py [-raw] file.dat|file.etl [file.dat|file.etl ...]
+
+Options:
+
+* `-raw` - Parse trace events and dump into <trace-file>.csv file.
+
+Output:
+
+* `<trace-file>.json.gz` - visualized driver activity, open in `<chrome://tracing/>` or `<edge://tracing/>`
+* `<trace-file>_stat.csv` - statistic of driver activity, open in Excel
+* `<trace-file>_surface.csv` - driver activity in surface point of view, open in Excel
+
+
+## Trace tool manifests and modules
+
+Manifests are for decoding trace data. Each trace system supported by trace tool need have a manifest file in manifests folder.
+The manifest file could be in MSFT ETW XML manifest format or json format. Current supported traces:
+* libva_trace.man        - libva trace manifest in MSFT ETW XML
+* Intel-Media-Open.json  - iHD open source media driver trace manifest in json.
+
+Trace tool dynamic load functional module in modules folders. Each module should have its own standalone function.
+If information exchange required among modules, should share through shared context, should not assume counter part module is available.
+current modules:
+* ftrace.py             - Linux ftace file parser, trace file from trace-cmd.
+* i915.py               - i915 trace handler to extract GPU workload submit&execution timing.
+* libva.py              - libva trace handler.
+* media.py              - Intel iHD open source media driver trace handler.
+* mediaRTLog.py         - Intel iHD open source media driver runtime log trace handler.
```

Comment:
they are split by function, mediaRTLog only handle runtime log event in media trace, format them into more user friendly format. RT log is auto generated format, not friendly for user.
all runtime log will be process and reformat by this module, so that user don't need pay attention to the detail of rt log.